lib/ansible/module_utils/basic.py

```diff
@@ -2020,15 +2031,15 @@ def run_command(self, args, check_rc=False, close_fds=True, executable=None, dat
         # create a printable version of the command for use
         # in reporting later, which strips out things like
         # passwords from the args list
-        if isinstance(args, basestring):
-            if isinstance(args, unicode):
-                b_args = args.encode('utf-8')
-            else:
-                b_args = args
-            to_clean_args = shlex.split(b_args)
-            del b_args
+        to_clean_args = args
+        if PY2:
+            if isinstance(args, text_type):
+                to_clean_args = args.encode('utf-8')
         else:
-            to_clean_args = args
+            if isinstance(args, binary_type):
+                to_clean_args = args.decode('utf-8', errors='replace')
+        if isinstance(args, (text_type, binary_type)):
+            to_clean_args = shlex.split(to_clean_args)
```

Comment:
splitter is much more complex than this code.  Since shlex has always worked here I'd rather not.  We should attempt to move as much as possible away from splitter rather than to it.